drivers/counter/Kconfig.mspm0

```diff
@@ -0,0 +1,6 @@
+config COUNTER_MSPM0
+	depends on DT_HAS_TI_COUNTER_MSPM0_ENABLED
```

Comment:
Formatting is off (my suggestion, sorry)
Also the convention seems to define the type first and then relations (see other drivers for order)